### PR TITLE
Add changes from pull 3639, workaround for GH #6211

### PR DESCRIPTION
**Pull request by soaxelbrooke**
This PR doesn't have the unit tests mentioned in GH #6211, but does act as a workaround for people who need to solve this problem before said unit tests are implemented.

### CODE REVIEW

**Comment by fchollet** [comment]
Please add a unit test, both for test coverage and for verifying the need for this feature (and its correctness).

File: keras/utils/generic_utils.py [hunk 1]
```diff
@@ -200,12 +200,28 @@ def func_load(code, defaults=None, closure=None, globs=None):
...
+    if closure is not None:
+        closure = func_reconstruct_closure(closure)
     return python_types.FunctionType(code, globs,
                                      name=code.co_name,
                                      argdefs=defaults,
                                      closure=closure)
 
 
+def func_reconstruct_closure(values):
+    """Deserialization helper that reconstructs a closure."""
```
Comment (fchollet) [inline keras/utils/generic_utils.py]: Please add `# Arguments` and `# Returns` sections.

Comment (philayres) [reply keras/utils/generic_utils.py]: I can confirm that this is a working solution and the code is reasonable.

I can't edit the changes, but for a simple copy and paste here are the arguments and returns as requested. Any chance we can get this incorporated without additional effort from @soaxelbrooke ?

```
    # Arguments
        closure: closure of the function.

    # Returns
        A reconstructed closure for func_load
```

**Comment by fchollet** [comment]
Checking back on this. Seems like got stalled due to expected edits that didn't come.

- Is this still an issue?
- If yes, is this PR still relevant to the issue?

Thanks!

**Comment by fchollet** [comment]
Closing; please reopen a PR if this is still a concern.

**Comment by mike-grayhat** [comment]
This bug still exists in tf 1.4.1

I think that the implementaion can be refined this way:

```python
def make_closure_cell(value):
    return (lambda: value).__closure__[0]
```

and in `utils.func_load`:

```python
if isinstance(closure, list):
    closure = tuple([make_closure_cell(cell_val) for cell_val in closure])
```

**Comment by philayres** [comment]
@mike-grayhat - out of interest, why is this better than the originally proposed PR? I will consider resurrecting this myself as a new PR. I'd like to know what the motivation is for your proposed approach so I'm not wasting my time on the original if it is broken in some way.

**Comment by mike-grayhat** [comment]
It seems cleaner (at least to me). Also using exec for putting a function (called func in this case) in the global namespace doesn't look like a very robust solution.

**Comment by philayres** [comment]
Fair enough. I can't really tell without more careful analysis whether your solution and the original actually have the same result. So based on this, if you'd like to own this PR I'm fine. I'll happily give your solution a test and give you feedback if that would be helpful.

**Comment by mike-grayhat** [comment]
I just checked - it's already in master and the latest tf release.

**Comment by philayres** [comment]
That's good to hear. Strange that the PR was rejected but the code made it in anyway. I don't follow the development closely, so I guess I shouldn't find it strange.

**Pull request closed without merging**